File: Models/DSSAT_EPICST_standalone/src/cs/DSSAT_EPICST_standalone/STEMP_EPIC_Component.cs
public class STEMP_EPIC_Component
{
    
        public STEMP_EPIC_Component() { }
    

    //Declaration of the associated strategies
    STEMP_EPIC _STEMP_EPIC = new STEMP_EPIC();

    public double[] BD
    {
        get
        {
             return _STEMP_EPIC.BD; 
        }
        set
        {
            _STEMP_EPIC.BD = value;
        }
    }
    public double[] DUL
    {
        get
        {
             return _STEMP_EPIC.DUL; 
        }
        set
        {
            _STEMP_EPIC.DUL = value;
        }
    }
    public double[] DS
    {
        get
        {
             return _STEMP_EPIC.DS; 
        }
        set
        {
            _STEMP_EPIC.DS = value;
        }
    }
    public double[] DLAYR
    {
        get
        {
             return _STEMP_EPIC.DLAYR; 
        }
        set
        {
            _STEMP_EPIC.DLAYR = value;
        }
    }
    public double[] LL
    {
        get
        {
             return _STEMP_EPIC.LL; 
        }
        set
        {
            _STEMP_EPIC.LL = value;
        }
    }
    public double[] SW
    {
        get
        {
             return _STEMP_EPIC.SW; 
        }
        set
        {
            _STEMP_EPIC.SW = value;
        }
    }
    public int NLAYR
    {
        get
        {
             return _STEMP_EPIC.NLAYR; 
        }
        set
        {
            _STEMP_EPIC.NLAYR = value;
        }
    }
    public int NL
    {
        get
        {
             return _STEMP_EPIC.NL; 
        }
        set
        {
            _STEMP_EPIC.NL = value;
        }
    }
    public string ISWWAT
    {
        get
        {
             return _STEMP_EPIC.ISWWAT; 
        }
        set
        {
            _STEMP_EPIC.ISWWAT = value;
        }
    }

    public void  CalculateModel(STEMP_EPIC_State s, STEMP_EPIC_State s1, STEMP_EPIC_Rate r, STEMP_EPIC_Auxiliary a, STEMP_EPIC_Exogenous ex)
    {
        _STEMP_EPIC.CalculateModel(s,s1, r, a, ex);
    }
    
    public STEMP_EPIC_Component(STEMP_EPIC_Component toCopy): this() // copy constructor 
    {

        
            for (int i = 0; i < 100; i++)
            { BD[i] = toCopy.BD[i]; }
    
        
            for (int i = 0; i < 100; i++)
            { DUL[i] = toCopy.DUL[i]; }
    
        
            for (int i = 0; i < 100; i++)
            { DS[i] = toCopy.DS[i]; }
    
        
            for (int i = 0; i < 100; i++)
            { DLAYR[i] = toCopy.DLAYR[i]; }
    
        
            for (int i = 0; i < 100; i++)
            { LL[i] = toCopy.LL[i]; }
    
        
            for (int i = 0; i < 100; i++)
            { SW[i] = toCopy.SW[i]; }
    
        NLAYR = toCopy.NLAYR;
        NL = toCopy.NL;
        ISWWAT = toCopy.ISWWAT;
    }
}

File: Models/DSSAT_EPICST_standalone/src/cpp1/DSSAT_EPICST_standalone/STEMP_EPIC_Component.cpp
#include "STEMP_EPIC_Component.h"

    STEMP_EPIC_Component::STEMP_EPIC_Component()
    {
           
    }
    

vector<double> & STEMP_EPIC_Component::getDUL() {return this-> DUL; }
string STEMP_EPIC_Component::getISWWAT() {return this-> ISWWAT; }
vector<double> & STEMP_EPIC_Component::getLL() {return this-> LL; }
vector<double> & STEMP_EPIC_Component::getDS() {return this-> DS; }
vector<double> & STEMP_EPIC_Component::getSW() {return this-> SW; }
vector<double> & STEMP_EPIC_Component::getBD() {return this-> BD; }
int STEMP_EPIC_Component::getNLAYR() {return this-> NLAYR; }
int STEMP_EPIC_Component::getNL() {return this-> NL; }
vector<double> & STEMP_EPIC_Component::getDLAYR() {return this-> DLAYR; }

void STEMP_EPIC_Component::setDUL(const vector<double> & _DUL)
{
    _STEMP_EPIC.setDUL(_DUL);
}
void STEMP_EPIC_Component::setISWWAT(string _ISWWAT)
{
    _STEMP_EPIC.setISWWAT(_ISWWAT);
}
void STEMP_EPIC_Component::setLL(const vector<double> & _LL)
{
    _STEMP_EPIC.setLL(_LL);
}
void STEMP_EPIC_Component::setDS(const vector<double> & _DS)
{
    _STEMP_EPIC.setDS(_DS);
}
void STEMP_EPIC_Component::setSW(const vector<double> & _SW)
{
    _STEMP_EPIC.setSW(_SW);
}
void STEMP_EPIC_Component::setBD(const vector<double> & _BD)
{
    _STEMP_EPIC.setBD(_BD);
}
void STEMP_EPIC_Component::setNLAYR(int _NLAYR)
{
    _STEMP_EPIC.setNLAYR(_NLAYR);
}
void STEMP_EPIC_Component::setNL(int _NL)
{
    _STEMP_EPIC.setNL(_NL);
}
void STEMP_EPIC_Component::setDLAYR(const vector<double> & _DLAYR)
{
    _STEMP_EPIC.setDLAYR(_DLAYR);
}
void STEMP_EPIC_Component::Calculate_Model(STEMP_EPIC_State& s, STEMP_EPIC_State& s1, STEMP_EPIC_Rate& r, STEMP_EPIC_Auxiliary& a, STEMP_EPIC_Exogenous& ex)
{
    _STEMP_EPIC.Calculate_Model(s, s1, r, a, ex);
}
STEMP_EPIC_Component::STEMP_EPIC_Component(const STEMP_EPIC_Component& toCopy)
{
    
        for (int i = 0; i < NL; i++)
        {
            DUL[i] = toCopy.DUL[i];
        }
    
    ISWWAT = toCopy.ISWWAT;
    
        for (int i = 0; i < NL; i++)
        {
            LL[i] = toCopy.LL[i];
        }
    
    
        for (int i = 0; i < NL; i++)
        {
            DS[i] = toCopy.DS[i];
        }
    
    
        for (int i = 0; i < NL; i++)
        {
            SW[i] = toCopy.SW[i];
        }
    
    
        for (int i = 0; i < NL; i++)
        {
            BD[i] = toCopy.BD[i];
        }
    
    NLAYR = toCopy.NLAYR;
    NL = toCopy.NL;
    
        for (int i = 0; i < NL; i++)
        {
            DLAYR[i] = toCopy.DLAYR[i];
        }
    
}